universal/ubi9/Dockerfile

```diff
@@ -0,0 +1,430 @@
+ENV MAVEN_HOME="/home/tooling/.sdkman/candidates/maven/current"
+
+ENV GRAALVM_HOME=/home/tooling/.sdkman/candidates/java/23.1.2.r21-mandrel
+
+ENV PATH="/home/tooling/.krew/bin:$PATH"
+ENV PATH="/home/tooling/.sdkman/candidates/maven/current/bin:$PATH"
+ENV PATH="/home/tooling/.sdkman/candidates/java/current/bin:$PATH"
+ENV PATH="/home/tooling/.sdkman/candidates/gradle/current/bin:$PATH"
+ENV PATH="/home/tooling/.local/share/coursier/bin:$PATH"
+
+# NodeJS
+RUN mkdir -p /home/tooling/.nvm/
+ENV NVM_DIR="/home/tooling/.nvm"
+ENV NODEJS_20_VERSION=20.18.0
+ENV NODEJS_18_VERSION=18.19.1
+ENV NODEJS_DEFAULT_VERSION=${NODEJS_18_VERSION}
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.5/install.sh | PROFILE=/dev/null bash
+RUN echo 'export NVM_DIR="$HOME/.nvm"' >> ${PROFILE_EXT} \
+    && echo '[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"' >> ${PROFILE_EXT}
+RUN source /home/user/.bashrc && \
+    nvm install v${NODEJS_20_VERSION} && \
+    nvm install v${NODEJS_18_VERSION} && \
+    nvm alias default v${NODEJS_DEFAULT_VERSION} && nvm use v${NODEJS_DEFAULT_VERSION} && \
+    npm install --global yarn@v1.22.17 &&\
+    chgrp -R 0 /home/tooling && chmod -R g=u /home/tooling
+ENV PATH=$NVM_DIR/versions/node/v${NODEJS_DEFAULT_VERSION}/bin:$PATH
+ENV NODEJS_HOME_20=$NVM_DIR/versions/node/v${NODEJS_20_VERSION}
+ENV NODEJS_HOME_18=$NVM_DIR/versions/node/v${NODEJS_18_VERSION}
+
+# kube
+ENV KUBECONFIG=/home/user/.kube/config
+
+USER 0
+
+# Required packages for AWT
+RUN dnf install -y libXext libXrender libXtst libXi
+
+# Lombok
+ENV LOMBOK_VERSION=1.18.18
+RUN wget -O /usr/local/lib/lombok.jar https://projectlombok.org/downloads/lombok-${LOMBOK_VERSION}.jar
+
+# Scala
+RUN curl -fLo cs https://git.io/coursier-cli && \
+    chmod +x cs && \
+    mv cs /usr/local/bin/
+RUN curl -fLo sbt https://raw.githubusercontent.com/dwijnand/sbt-extras/master/sbt && \
+    chmod +x sbt && \
+    mv sbt /usr/local/bin/
+RUN curl -fLo mill https://raw.githubusercontent.com/lefou/millw/main/millw && \
+    chmod +x mill && \
+    mv mill /usr/local/bin/
+
+# C/CPP
+RUN dnf -y install llvm-toolset gcc gcc-c++ clang clang-libs clang-tools-extra gdb
+
+# Go 1.18+    - installed to /usr/bin/go
+# gopls 0.10+ - installed to /home/tooling/go/bin/gopls and /home/tooling/go/pkg/mod/
+RUN dnf install -y go-toolset && \
+    GO111MODULE=on go install -v golang.org/x/tools/gopls@latest && \
+    chgrp -R 0 /home/tooling && chmod -R g=u /home/tooling
+ENV GOBIN="/home/tooling/go/bin/"
+ENV PATH="$GOBIN:$PATH"
+
+# Python
+RUN dnf -y install python3.11 python3.11-devel python3.11-setuptools python3.11-pip nss_wrapper
+
+RUN cd /usr/bin \
+    && if [ ! -L python ]; then ln -s python3.11 python; fi \
+    && if [ ! -L pydoc ]; then ln -s pydoc3.11 pydoc; fi \
+    && if [ ! -L python-config ]; then ln -s python3.11-config python-config; fi \
+    && if [ ! -L pip ]; then ln -s pip-3.11 pip; fi
+
+RUN pip install pylint yq
+
+# PHP
+ENV PHP_VERSION=8.1
```

Comment:
I want to point out that for the ubi8 based udi, the PHP_VERSION is 7.4:
https://github.com/devfile/developer-images/blob/0eb879a7fccd12da128f89b99a986292a0b9091a/universal/ubi8/Dockerfile#L140

But for ubi9, the supported streams by default are:
```
Name Stream Profiles               Summary               
php  8.1    common, devel, minimal PHP scripting language
php  8.2    common, devel, minimal PHP scripting language
```

so I've updated the PHP_VERSION to 8.1.

Hello @svor , is there a strong reason for the UDI to stay on PHP 7.4?

Reply:
hi @dkwon17, I don't it's a problem having PHP 8.1 instead of 7.4

Reply:
Thank you for the clarification. In my latest commits, I've updated the version to 8.2 which is the latest available stream